tg_bot/modules/muting.py

```diff
@@ -0,0 +1,243 @@
+    message = update.effective_message  # type: Optional[Message]
+
+    user_id, reason = extract_user_and_text(message, args)
+
+    if not user_id:
+        message.reply_text("Non ti stai riferendo ad un utente")
+        return ""
+
+    try:
+        member = chat.get_member(user_id)
+    except BadRequest as excp:
+        if excp.message == "User not found":
+            message.reply_text("Non riesco a trovare questo utente")
+            return ""
+        else:
+            raise
+
+    if is_user_admin(chat, user_id, member):
+        message.reply_text("Mi piacerebbe tanto poter mutare gli admin a volte...")
+        return ""
+
+    if user_id == bot.id:
+        message.reply_text("Non mi muterò da solo.. Sei pazzo?")
+        return ""
+
+    if not reason:
+        message.reply_text("NOn hai specificato per quanto tempo mutare questo utente!")
+        return ""
+
+    split_reason = reason.split(None, 1)
+
+    time_val = split_reason[0].lower()
+    if len(split_reason) > 1:
+        reason = split_reason[1]
+    else:
+        reason = ""
+
+    mutetime = extract_time(message, time_val)
+
+    if not mutetime:
+        return ""
+
+    log = (
+        "<b>{}:</b>"
+        "\n#TEMP MUTED"
+        "\n<b>Admin:</b> {}"
+        "\n<b>User:</b> {}"
+        "\n<b>Time:</b> {}".format(
+            html.escape(chat.title),
+            mention_html(user.id, user.first_name),
+            mention_html(member.user.id, member.user.first_name),
+            time_val,
+        )
+    )
+    if reason:
+        log += "\n<b>Motivo:</b> {}".format(reason)
+
+    try:
+        if member.can_send_messages is None or member.can_send_messages:
+            bot.restrict_chat_member(
+                chat.id, user_id, until_date=mutetime, can_send_messages=False
+            )
+            message.reply_text("Mutato per {}!".format(time_val))
+            return log
+        else:
+            message.reply_text("Questo utente è già mutato.")
+
+    except BadRequest as excp:
+        if excp.message == "Reply message not found":
+            # Do not reply
+            message.reply_text("Mutato per {}!".format(time_val), quote=False)
+            return log
+        else:
+            LOGGER.warning(update)
+            LOGGER.exception(
+                "ERROR muting user %s in chat %s (%s) due to %s",
+                user_id,
+                chat.title,
+                chat.id,
+                excp.message,
+            )
+            message.reply_text("Diamine, non posso mutare questo utente.")
+
+    return ""
+
+
+__help__ = """
+*Admin only:*
+ - /mute <userhandle>: silenzia un utente. Può anche essere usato come risposta, disattivando la risposta all'utente.
+ - /tmute <userhandle> x (m / h / d): silenzia un utente per x tempo. (tramite handle o risposta). m = minuti, h = ore, d = giorni.
+ - /unmute <userhandle>: riattiva un utente. Può anche essere usato come risposta, disattivando la risposta all'utente.
```

Comment:
Non sono sicuro di aver capito cosa si intende con "disattivando la risposta all'utente". Vuol dire che l'utente non può rispondere? Come mai è usato anche per l'unmute? Non dovrebbe essere una "riattivazione" in quel caso?

Reply:
credo voglia dire che l'utente a cui rispondi viene mutato... proviamo sul gruppo e casomai modifichiamo il testo di spiegazione

Reply:
> Non sono sicuro di aver capito cosa si intende con "disattivando la risposta all'utente". Vuol dire che l'utente non può rispondere? Come mai è usato anche per l'unmute? Non dovrebbe essere una "riattivazione" in quel caso?

Significa che rispondendo a quel messaggio muti l'utente, senza doverlo citare con "@"